File: app/src/main/java/com/mytech/mangatalkreader/ui/screens/manga_details/MangaDetailsScreen.java
package com.mytech.mangatalkreader.ui.screens.manga_details;

import androidx.compose.foundation.clickable;
import androidx.compose.foundation.layout.Arrangement;
import androidx.compose.foundation.layout.Box;
import androidx.compose.foundation.layout.Column;
import androidx.compose.foundation.layout.Row;
import androidx.compose.foundation.layout.Spacer;
import androidx.compose.foundation.layout.fillMaxSize;
import androidx.compose.foundation.layout.fillMaxWidth;
import androidx.compose.foundation.layout.height;
import androidx.compose.foundation.layout.padding;
import androidx.compose.foundation.layout.size;
import androidx.compose.foundation.layout.width;
import androidx.compose.foundation.lazy.LazyColumn;
import androidx.compose.foundation.lazy.items;
import androidx.compose.material.icons.Icons;
import androidx.compose.material.icons.filled.ArrowBack;
import androidx.compose.material.icons.filled.Favorite;
import androidx.compose.material.icons.filled.FavoriteBorder;
import androidx.compose.material3.Card;
import androidx.compose.material3.CircularProgressIndicator;
import androidx.compose.material3.ExperimentalMaterial3Api;
import androidx.compose.material3.HorizontalDivider;
import androidx.compose.material3.Icon;
import androidx.compose.material3.IconButton;
import androidx.compose.material3.MaterialTheme;
import androidx.compose.material3.Scaffold;
import androidx.compose.material3.Text;
import androidx.compose.material3.TopAppBar;
import androidx.compose.runtime.Composable;
import androidx.compose.runtime.collectAsState;
import androidx.compose.runtime.getValue;
import androidx.compose.runtime.mutableStateListOf;
import androidx.compose.runtime.mutableStateOf;
import androidx.compose.runtime.remember;
import androidx.compose.runtime.setValue;
import androidx.compose.ui.Alignment;
import androidx.compose.ui.Modifier;
import androidx.compose.ui.layout.ContentScale;
import androidx.compose.ui.platform.LocalContext;
import androidx.compose.ui.text.font.FontWeight;
import androidx.compose.ui.text.style.TextOverflow;
import androidx.compose.ui.unit.dp;
import androidx.compose.ui.unit.sp;
import coil.compose.AsyncImage;
import coil.request.ImageRequest;
import com.mytech.mangatalkreader.data.db.entity.ChapterEntity;
import com.mytech.mangatalkreader.data.db.entity.MangaEntity;
import com.mytech.mangatalkreader.ui.viewmodel.MangaDetailsViewModel;
import java.util.Collections;
import java.util.List;

/**
 * Manga details screen - shows manga info and chapter list.
 * Real data from database, no stubs.
 */
public final class MangaDetailsScreenKt {

    @OptIn(ExperimentalMaterial3Api.class)
    @Composable
    public static void MangaDetailsScreen(
            MangaDetailsViewModel viewModel
    ) {
        MangaEntity manga = viewModel.getManga().collectAsState(null).getValue();
        List<ChapterEntity> chapters = viewModel.getChapters().collectAsState(Collections.emptyList()).getValue();
        
        Scaffold(
            topBar = () -> TopAppBar(
                title = () -> Text(manga != null ? manga.getTitle() : "Manga Details"),
                navigationIcon = () -> IconButton(onClick = () -> {}),
                actions = () -> {
                    if (manga != null) {
                        IconButton(
                            onClick = () -> {
                                // Toggle favorite
                                MangaEntity updated = manga.copy(
                                    manga.getId(),
                                    manga.getMangaId(),
                                    manga.getName(),
                                    manga.getNumber(),
                                    manga.getUrl(),
                                    manga.getPath(),
                                    manga.getCurrentPage(),
                                    !manga.isRead(),
                                    manga.getDateAdded(),
                                    manga.getSource(),
                                    manga.getPagesCount(),
                                    manga.getDateUpload(),
                                    manga.getTitle()
                                );
                                viewModel.updateManga(updated);
                            }
                        );
                    }
                }
            ),
            content = innerPadding -> {
                if (manga == null) {
                    Box(
                        Modifier.fillMaxSize().padding(innerPadding),
                        contentAlignment = Alignment.Center
                    ) {
                        CircularProgressIndicator();
                    }
                } else {
                    LazyColumn(
                        modifier = Modifier.fillMaxSize().padding(innerPadding)
                    ) {
                        item {
                            MangaHeader(manga);
                        }
                        item {
                            HorizontalDivider();
                        }
                        items(chapters, key = chapter -> chapter.getId()) { chapter ->
                            ChapterItem(chapter);
                        }
                    }
                }
            }
        );
    }

    @Composable
    private static void MangaHeader(MangaEntity manga) {
        Column(
            modifier = Modifier.fillMaxWidth().padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            if (manga.getCoverUrl() != null && !manga.getCoverUrl().isEmpty()) {
                AsyncImage(
                    model = ImageRequest.Builder(LocalContext.get())
                        .data(manga.getCoverUrl())
                        .crossfade(true)
                        .build(),
                    contentDescription = manga.getTitle(),
                    modifier = Modifier.fillMaxWidth().height(250.dp),
                    contentScale = ContentScale.Crop
                );
            }
            
            Text(
                text = manga.getTitle(),
                fontSize = 24.sp,
                fontWeight = FontWeight.Bold
            );
            
            if (manga.getAuthor() != null && !manga.getAuthor().isEmpty()) {
                Text(
                    text = "Автор: " + manga.getAuthor(),
                    fontSize = 14.sp,
                    color = MaterialTheme.INSTANCE.getColorScheme().getOnSurfaceVariant()
                );
            }
            
            if (manga.getDescription() != null && !manga.getDescription().isEmpty()) {
                Text(
                    text = manga.getDescription(),
                    fontSize = 14.sp,
                    maxLines = 4,
                    overflow = TextOverflow.Ellipsis
                );
            }
        }
    }

    @Composable
    private static void ChapterItem(ChapterEntity chapter) {
        Card(
            modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 4.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth().padding(16.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = chapter.getName(),
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Medium
                    );
                    Text(
                        text = "Страниц: " + chapter.getPagesCount(),
                        fontSize = 12.sp,
                        color = MaterialTheme.INSTANCE.getColorScheme().getOnSurfaceVariant()
                    );
                }
                
                if (chapter.isRead()) {
                    Icon(
                        Icons.Default.Favorite,
                        contentDescription = "Read",
                        tint = MaterialTheme.INSTANCE.getColorScheme().getPrimary()
                    );
                }
            }
        }
    }
}

File: app/src/main/java/com/mytech/mangatalkreader/ui/screens/mangadetails/MangaDetailsScreen.kt
package com.mytech.mangatalkreader.ui.screens.mangadetails

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FavoriteBorder
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import coil.compose.AsyncImage
import coil.request.ImageRequest
import com.mytech.mangatalkreader.data.db.entity.ChapterEntity
import com.mytech.mangatalkreader.data.db.entity.MangaEntity
import com.mytech.mangatalkreader.ui.viewmodel.MangaDetailsViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MangaDetailsScreen(
    mangaId: Long,
    onChapterClick: (Long) -> Unit,
    onNavigateBack: () -> Unit,
    modifier: Modifier = Modifier,
    viewModel: MangaDetailsViewModel = hiltViewModel()
) {
    val manga by viewModel.manga.collectAsStateWithLifecycle()
    val chapters by viewModel.chapters.collectAsStateWithLifecycle()

    Scaffold(
        modifier = modifier,
        topBar = {
            TopAppBar(
                title = { Text(manga?.title ?: "Manga Details") },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(
                            imageVector = androidx.compose.material.icons.Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                },
                actions = {
                    IconButton(onClick = { viewModel.toggleFavorite() }) {
                        Icon(
                            imageVector = if (manga?.isFavorite == true) Icons.Default.Favorite else Icons.Default.FavoriteBorder,
                            contentDescription = "Toggle favorite",
                            tint = if (manga?.isFavorite == true) MaterialTheme.colorScheme.error
                            else MaterialTheme.colorScheme.onSurface
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.surface.copy(alpha = 0.95f)
                )
            )
        }
    ) { paddingValues ->
        when {
            manga == null && viewModel.isLoading -> {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(paddingValues),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            }

            manga == null -> {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(paddingValues),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "Manga not found",
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.error
                    )
                }
            }

            else -> {
                LazyColumn(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(paddingValues),
                    contentPadding = PaddingValues(16.dp),
                    verticalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    item {
                        MangaHeader(manga = manga!!)
                    }

                    item {
                        Text(
                            text = "Chapters (${chapters.size})",
                            style = MaterialTheme.typography.titleLarge,
                            fontWeight = FontWeight.Bold
                        )
                    }

                    items(chapters, key = { it.id }) { chapter ->
                        ChapterItem(
                            chapter = chapter,
                            onClick = { onChapterClick(chapter.id) }
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun MangaHeader(
    manga: MangaEntity,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier) {
        Row {
            if (manga.localCoverPath != null && manga.localCoverPath.isNotBlank()) {
                AsyncImage(
                    model = ImageRequest.Builder(LocalContext.current)
                        .data(manga.localCoverPath)
                        .crossfade(true)
                        .build(),
                    contentDescription = manga.title,
                    modifier = Modifier
                        .width(120.dp)
                        .aspectRatio(0.7f)
                        .clip(RoundedCornerShape(8.dp)),
                    contentScale = ContentScale.Crop
                )
            } else if (manga.coverUrl.isNotBlank()) {
                AsyncImage(
                    model = ImageRequest.Builder(LocalContext.current)
                        .data(manga.coverUrl)
                        .crossfade(true)
                        .build(),
                    contentDescription = manga.title,
                    modifier = Modifier
                        .width(120.dp)
                        .aspectRatio(0.7f)
                        .clip(RoundedCornerShape(8.dp)),
                    contentScale = ContentScale.Crop
                )
            }

            Spacer(modifier = Modifier.width(12.dp))

            Column {
                Text(
                    text = manga.title,
                    style = MaterialTheme.typography.headlineSmall,
                    fontWeight = FontWeight.Bold,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis
                )

                if (manga.author.isNotBlank()) {
                    Text(
                        text = manga.author,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }

                Text(
                    text = manga.status.replaceFirstChar { it.uppercase() },
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.secondary
                )

                if (manga.genre.isNotBlank()) {
                    Text(
                        text = manga.genre.split(",").filter { it.isNotBlank() }.take(5).joinToString(", "),
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.tertiary,
                        maxLines = 2,
                        overflow = TextOverflow.Ellipsis
                    )
                }
            }
        }

        if (manga.description.isNotBlank()) {
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = manga.description,
                style = MaterialTheme.typography.bodyMedium,
                maxLines = 5,
                overflow = TextOverflow.Ellipsis
            )
        }
    }
}

@Composable
private fun ChapterItem(
    chapter: ChapterEntity,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.clickable(onClick = onClick),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
        shape = RoundedCornerShape(8.dp)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = if (chapter.title.isNotBlank()) {
                        chapter.title
                    } else {
                        "Chapter ${chapter.chapterNumber}"
                    },
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.SemiBold,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )

                if (chapter.scanlator.isNotBlank()) {
                    Text(
                        text = chapter.scanlator,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            Icon(
                imageVector = Icons.Default.PlayArrow,
                contentDescription = "Read chapter",
                tint = MaterialTheme.colorScheme.primary,
                modifier = Modifier.size(24.dp)
            )
        }
    }
}
